Abigen produces bindings with dependency issues

**Version**
439a0c7de0fb8569178d56933b000d8783805c2f

**Platform**
MacOS

**Description**
https://github.com/gakonst/ethers-rs/pull/2222 seems to have affected binding generation. `forge bind` generates bindings which produce the following errors 

```
error[E0432]: unresolved import `futures_util::try_join`
   |
35 | use futures_util::{lock::Mutex, try_join};
   |                                 ^^^^^^^^ no `try_join` in the root

error: cannot determine resolution for the macro `try_join`
    |
789 |             try_join!(self.resolve_field(ens_name, "avatar"), self.resolve_name(ens_name))?;
    |             ^^^^^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports

    Checking ethers-contract-abigen v2.0.0 (https://github.com/gakonst/ethers-rs#439a0c7d)
```

Relatedly, should the generated `Cargo.toml` file pin ethers to a specific version?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,6 +58,8 @@ default-members = [
     "ethers-solc",
 ]
 
+# Explicitly set the resolver to version 2, which is the default for packages with edition >= 2021
+# https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html
 resolver = "2"
 
 [workspace.dependencies]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,6 +58,8 @@ default-members = [
     "ethers-solc",
 ]
 
+resolver = "2"
+
 [workspace.dependencies]
 # workspace crates
 ethers = { version = "2.0.0", path = "ethers", default-features = false }
@@ -77,11 +79,11 @@ ethers-derive-eip712 = { version = "2.0.0", path = "ethers-core/ethers-derive-ei
 # async / async utils
 tokio = "1.26"
 tokio-tungstenite = { version = "0.18.0", default-features = false }
-futures = { version = "0.3.27", default-features = false }
-futures-core = { version = "0.3.27", default-features = false }
-futures-util = { version = "0.3.27", default-features = false }
-futures-executor = { version = "0.3.27", default-features = false }
-futures-channel = { version = "0.3.27", default-features = false }
+futures = { version = "0.3.27", default-features = false, features = ["std"] }
+futures-core = "0.3.27"
+futures-util = "0.3.27"
+futures-executor = "0.3.27"
+futures-channel = "0.3.27"
 futures-locks = { version = "0.7.1", default-features = false }
 futures-timer = { version = "3.0.2", default-features = false, features = ["wasm-bindgen"] }
 pin-project = "1.0"
@@ -103,7 +105,7 @@ serde_json = "1.0"
 # macros
 proc-macro2 = "1.0"
 quote = "1.0"
-syn = "1.0"
+syn = { version = "1.0", features = ["extra-traits"] }
 async-trait = "0.1.66"
 auto_impl = "1.0"
 

--- a/ethers-contract/ethers-contract-abigen/Cargo.toml
+++ b/ethers-contract/ethers-contract-abigen/Cargo.toml
@@ -30,7 +30,7 @@ ethers-core = { workspace = true, features = ["macros"] }
 
 proc-macro2.workspace = true
 quote.workspace = true
-syn = { workspace = true, features = ["full"] }
+syn.workspace = true
 prettyplease = "0.1.25"
 
 Inflector = "0.11"

